packages/core/src/lib/passcode.ts

```diff
@@ -30,3 +31,19 @@ export const createPasscode = async (
     ...payload,
   });
 };
+
+export const sendPasscode = async (passcode: Passcode) => {
+  const emailOrPhone = passcode.email ?? passcode.phone;
+
+  if (!emailOrPhone) {
+    throw new Error('Both email and phone are empty.');
```

Comment:
Should be a request error?

Reply:
is more like a system misconfigure error (500)

this kind of passcode is not expect to exist

Reply:
will be refactored in the very next PR(drafted) to avoid conflict.